.github/workflows/connector-ci-tests.yml

```diff
@@ -1,5 +1,9 @@
 name: Connector CI Tests
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```

Comment:
Could we set this at the matrix level, so the free runners can still complete without interruption?

Reply:
That's the tradeoff. I'm proposing always killing the previous tests is better since the current commit should run (and we need the current commit to be green to merge anyway)

To me, the main benefit to leaving the previous test running is if someone is trying to debug a test and diffing test output across different commits. That seems rare and.. inefficient.

Reply:
I know you are busy and I don't want to block. I have a new PR that just moves this `cancel-in-progress` config into the Java test matrix. This respects both of our 
expressed priorities (as I understand them), with no down-sides I can see.

Reply:
- https://github.com/airbytehq/airbyte/pull/60277

Reply:
🤔 I hear you. We used to follow this model on the Platform side, and moved to cancel-in-progress because we found the guarantees it provided weren't as great in practice, and devs often relied on the latest commit anyway i.e. if I'm rapidly developing, highly likely I have logic changes that aren't easy to revert. My understanding is this is why previous dagger pipeline also cancelled in progress test - have you heard feedback otherwise?